packages/fuchsia_remote_debug_protocol/lib/src/dart/dart_vm.dart

```diff
@@ -0,0 +1,180 @@
+      return peer;
+    } catch (e) {
+      await peer?.close();
+      await socket?.close();
+      if (timer.elapsed < _kConnectTimeout) {
+        _log.info('Attempting to reconnect');
+        await new Future<Null>.delayed(_kReconnectAttemptInterval);
+        return attemptConnection(uri);
+      } else {
+        _log.severe('Connection to Fuchsia\'s Dart VM timed out at '
+            '${uri.toString()}');
+        rethrow;
+      }
+    }
+  }
+
+  return attemptConnection(uri);
+}
+
+/// Restores the VM service connection function to the original state.
+void restoreVmServiceConnectionFunction() {
+  fuchsiaVmServiceConnectionFunction = _waitAndConnect;
+}
+
+/// Simple JSON RPC peer with a Fuchsia Dart VM service instance. Wraps most
+/// methods around a JSON RPC peer class.
+class DartVm {
+  final json_rpc.Peer _peer;
+
+  DartVm._(this._peer);
+
+  /// Attempts to connect to the given `Uri`.
+  ///
+  /// Throws an error if unable to connect.
+  static Future<DartVm> connect(Uri uri) async {
+    if (uri.scheme == 'http') {
+      uri = uri.replace(scheme: 'ws', path: '/ws');
+    }
+    final json_rpc.Peer peer = await fuchsiaVmServiceConnectionFunction(uri);
+    if (peer == null) {
+      return null;
+    }
+    return new DartVm._(peer);
+  }
+
+  /// Invokes a raw JSON RPC command with the VM service.
+  ///
+  /// When `timeout` is set and reached, throws a `TimeoutException`.
+  ///
+  /// If the function returns, it is with a raw JSON response.
+  Future<Map<String, dynamic>> invokeRpc(String function,
+      {Map<String, dynamic> params, Duration timeout}) async {
+    final Future<Map<String, dynamic>> future =
+        _peer.sendRequest(function, params ?? <String, dynamic>{});
+    if (timeout == null) {
+      return future;
+    }
+    return future.timeout(timeout, onTimeout: () {
+      throw new TimeoutException(
+          'Peer connection timed out during RPC call', timeout);
+    });
+  }
+
+  /// Returns a list of `FlutterView`s running across all Dart VM's.
+  ///
+  /// If there is no associated isolate with the flutter view (used to determine
+  /// the flutter view's name), then the flutter view's ID will be added
+  /// instead. If none of these things can be found (isolate has no name or the
+  /// flutter view has no ID), then the result will not be added to the list.
+  Future<List<FlutterView>> getAllFlutterViews() async {
+    final List<FlutterView> views = <FlutterView>[];
+    final Map<String, dynamic> rpcResponse =
+        await invokeRpc('_flutter.listViews', timeout: _kRpcTimeout);
+    final List<Map<String, dynamic>> flutterViewsJson = rpcResponse['views'];
+    for (Map<String, dynamic> jsonView in flutterViewsJson) {
+      final FlutterView flutterView = new FlutterView._fromJson(jsonView);
+      if (flutterView != null) {
+        views.add(flutterView);
+      }
+    }
+    return views;
+  }
+
+  /// Shuts down all active connections.
```

Comment:
"Disconnects from the Dart VM service"? :) 

We should probably mention that this object is no longer usable after calling this method.

Reply:
Done.